src/platform/nxp/k32w/k32w1/FactoryDataProviderImpl.cpp

```diff
@@ -118,18 +159,27 @@ CHIP_ERROR FactoryDataProviderImpl::SSS_Sign(uint8_t * digest, Crypto::P256ECDSA
     return error;
 }
 
-#if CHIP_DEVICE_CONFIG_SECURE_DAC_PRIVATE_KEY
 CHIP_ERROR FactoryDataProviderImpl::SSS_ConvertDacKey()
 {
     size_t blobSize                     = kPrivateKeyBlobLength;
     size_t newSize                      = sizeof(FactoryDataProvider::Header) + mHeader.size + kSssBlobMetadataLength;
     uint8_t blob[kPrivateKeyBlobLength] = { 0 };
     uint8_t * data                      = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(newSize));
     uint32_t offset                     = 0;
+    bool convNeeded                     = true;
+    CHIP_ERROR error                    = CHIP_NO_ERROR;
 
     VerifyOrReturnError(data != nullptr, CHIP_ERROR_INTERNAL);
 
-    ReturnErrorOnFailure(SSS_ExportBlob(blob, &blobSize, offset));
+    error = SSS_ExportBlob(blob, &blobSize, offset, convNeeded);
+    if (!convNeeded)
+    {
+        ChipLogError(DeviceLayer, "SSS: DAC private key already converted to blob");
+        chip::Platform::MemoryFree(data);
+        return CHIP_NO_ERROR;
+    }
+
+    ReturnErrorOnFailure(error);
```

Comment:
I would suggest doing error checking before using values returned by a function. In your case it will work, but it looks like a potential bug.

Reply:
Thank you. I missed that